src/btree/bt_debug.c

```diff
@@ -717,6 +733,96 @@ __wt_debug_cursor_hs(void *cursor_arg, const char *ofile)
     return (__wt_debug_tree_all(hs_session, cbt->btree, NULL, ofile));
 }
 
+/*
+ * __wt_debug_cursor_hs --
+ *     Dump information about a single history store cursor.
+ */
+int
+__wt_debug_cursor_hs(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, bool dump_key,
+  bool dump_time_pairs, bool dump_value)
+{
+    WT_DBG *ds, _ds;
+    WT_DECL_ITEM(hs_key);
+    WT_DECL_ITEM(hs_value);
+    WT_DECL_RET;
+    WT_TIME_PAIR start, stop;
+    WT_UPDATE *upd;
+    wt_timestamp_t hs_durable_ts;
+    size_t notused;
+    uint32_t hs_btree_id;
+    uint8_t hs_prep_state, hs_upd_type;
+
+    ds = &_ds;
+    notused = 0;
+
+    WT_ERR(__wt_scr_alloc(session, 0, &hs_key));
+    WT_ERR(__wt_scr_alloc(session, 0, &hs_value));
+    WT_ERR(__debug_config(session, ds, NULL));
+
+    if (dump_key || dump_time_pairs)
+        WT_ERR(hs_cursor->get_key(hs_cursor, &hs_btree_id, hs_key, &start.timestamp, &start.txnid,
+          &stop.timestamp, &stop.txnid));
+
+    if (dump_key)
+        WT_ERR(__debug_item_key(ds, "K", hs_key->data, hs_key->size));
+
+    if (dump_time_pairs)
+        WT_ERR(
+          __debug_time_pairs(ds, "T", start.timestamp, start.txnid, stop.timestamp, stop.txnid));
+
+    if (dump_value) {
+        WT_ERR(
+          hs_cursor->get_value(hs_cursor, &hs_durable_ts, &hs_prep_state, &hs_upd_type, hs_value));
+        switch (hs_upd_type) {
+        case WT_UPDATE_MODIFY:
+            WT_ERR(__wt_update_alloc(session, hs_value, &upd, &notused, hs_upd_type));
+            WT_ERR(__debug_modify(ds, upd, "\tM "));
+            break;
+        case WT_UPDATE_STANDARD:
+            WT_ERR(__debug_item_value(ds, "V", hs_value->data, hs_value->size));
+            break;
+        default:
+            /*
```

Comment:
As this is being called within verify we should throw some kind of error here I think? That way when we arrive at an unexpected case we will fail.

Reply:
I'll add an assert